mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -206,11 +212,21 @@ object KMeansModel extends MLReadable[KMeansModel] {
       import sqlContext.implicits._
 
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
-
       val dataPath = new Path(path, "data").toString
-      val data: Dataset[Data] = sqlContext.read.parquet(dataPath).as[Data]
-      val clusterCenters = data.collect().sortBy(_.clusterIdx).map(_.clusterCenter)
-      val model = new KMeansModel(metadata.uid, new MLlibKMeansModel(clusterCenters))
+
+      val versionRegex = "([0-9]+)\\.(.+)".r
+      val versionRegex(major, _) = metadata.sparkVersion
+
+      val model = if (major.toInt >= 2) {
```

Comment:
@jkbradley It does not work as expected when spark version > 10.0.
For example, if ```metadata.sparkVersion = "10.0"```, ```metadata.sparkVersion >= "2.0"``` will return false which is not correct.
We can not use ```metadata.sparkVersion.toDouble``` to compare as well. For example, if ```metadata.sparkVersion = "1.6.1"```, ```metadata.sparkVersion.toDouble``` will throw the following  exception
```
java.lang.NumberFormatException: multiple points
```
So I think using regex is the most robust method.